### DISCUSSION TITLE
Confused about web_sys + clipboard usage

### DISCUSSION THREAD

**Discussion by miketwenty1** [wasm-bindgen/wasm-bindgen#3223] [Q&A] [answered]
### tl;dr 
looking for full example with Clipboard usage. I'm guessing something like this is the right approach? 
```rust
#[cfg(web_sys_unstable_apis)]
fn somefn(//..
) {
    // ..
    let window = web_sys::window().expect("window"); 
    let nav = window.navigator();
    if let Some(clipboard) = nav.clipboard() {
        let _ = clipboard.write_text("testingclipboard1234");
        info!("text written to clipboard maybe");
    } else {
        warn!("no navigator clipboard");
    }
    /..
}
```

<img width="844" alt="Screen Shot 2022-12-31 at 12 31 35 AM" src="https://user-images.githubusercontent.com/7761473/210126221-61aacef2-bb35-4352-ae8d-636689220a64.png">

But I'm unable to setup my rust analyzer to see the clipboard method.

### details:
I'm running a bevy wasm game and I would like to populate my users clipboard.
I'm guessing this would be the write/writetext methods for:
https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Clipboard.html
?

I believe it is correct if trying to use this I need to add this to my `Cargo.toml`:
```toml
[dependencies.web-sys]
version = "0.3"
features = ["Clipboard", "Window", "Navigator", "Permissions"]
```
`.cargo/config.toml`:
```toml
[target.wasm32-unknown-unknown]
runner = "wasm-server-runner"
[build]
rustflags = ["--cfg=web_sys_unstable_apis"]
```
The build flag doesn't seem to be working?

I'm leaving this off as it breaks my bevy app. 
```toml
[lib]
crate-type = ["cdylib"]
```
Not sure if this is needed.^ Thought I should note this because it's in every example and I'm not really sure why.

I also have in my rust file:
```
use wasm_bindgen::prelude::wasm_bindgen;
```

**Comment by miketwenty1** [comment, marked as answer]
I solved my issue:
```rust
use wasm_bindgen_futures::spawn_local;
let _task = spawn_local(async move {
    let window = web_sys::window().expect("window"); // { obj: val };
    let nav = window.navigator().clipboard();
    match nav {
        Some(a) => {
            let p = a.write_text("please god work");
            let result = wasm_bindgen_futures::JsFuture::from(p)
                .await
                .expect("clipboard populated");
            info!("clippyboy worked");
        }
        None => {
            warn!("failed to copy clippyboy");
        }
    };
});
```
